renew/promote.go

```diff
@@ -0,0 +1,204 @@
+package renew
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+)
+
+type PromoteReq struct {
+	Date string `json:"date"`
+}
+
+type PromoteResp struct {
+	Success []RenewReport `json:"success"`
+	Failure []RenewReport `json:"failure"`
+}
+
+type RenewReport struct {
+	Policy       models.Policy        `json:"policy"`
+	Transactions []models.Transaction `json:"transactions"`
+	Error        error                `json:"error,omitempty"`
+}
```

Comment:
should we creste a models file inside renew folder that will contain all these common structs